ql/math/matrix.hpp

```diff
@@ -217,502 +217,9 @@ namespace QuantLib {
-        QL_REQUIRE(i<columns_,
-                   "column index (" << i << ") must be less than " << columns_ <<
-                   ": matrix cannot be accessed out of range");
-        #endif
-        return const_column_iterator(data_.get()+i,columns_);
-    }
-
-    inline Matrix::column_iterator Matrix::column_begin(Size i) {
-        #if defined(QL_EXTRA_SAFETY_CHECKS)
-        QL_REQUIRE(i<columns_,
-                   "column index (" << i << ") must be less than " << columns_ <<
-                   ": matrix cannot be accessed out of range");
-        #endif
-        return column_iterator(data_.get()+i,columns_);
-    }
-
-    inline Matrix::const_column_iterator
-    Matrix::column_end(Size i) const {
-        #if defined(QL_EXTRA_SAFETY_CHECKS)
-        QL_REQUIRE(i<columns_,
-                   "column index (" << i << ") must be less than " << columns_ <<
-                   ": matrix cannot be accessed out of range");
-        #endif
-        return const_column_iterator(data_.get()+i+rows_*columns_,columns_);
-    }
-
-    inline Matrix::column_iterator Matrix::column_end(Size i) {
-        #if defined(QL_EXTRA_SAFETY_CHECKS)
-        QL_REQUIRE(i<columns_,
-                   "column index (" << i << ") must be less than " << columns_ <<
-                   ": matrix cannot be accessed out of range");
-        #endif
-        return column_iterator(data_.get()+i+rows_*columns_,columns_);
-    }
-
-    inline Matrix::const_reverse_column_iterator
-    Matrix::column_rbegin(Size i) const {
-        return const_reverse_column_iterator(column_end(i));
-    }
-
-    inline Matrix::reverse_column_iterator
-    Matrix::column_rbegin(Size i) {
-        return reverse_column_iterator(column_end(i));
-    }
-
-    inline Matrix::const_reverse_column_iterator
-    Matrix::column_rend(Size i) const {
-        return const_reverse_column_iterator(column_begin(i));
-    }
-
-    inline Matrix::reverse_column_iterator
-    Matrix::column_rend(Size i) {
-        return reverse_column_iterator(column_begin(i));
-    }
-
-    inline Matrix::const_row_iterator
-    Matrix::operator[](Size i) const {
-        return row_begin(i);
-    }
-
-    inline Matrix::const_row_iterator
-    Matrix::at(Size i) const {
-        QL_REQUIRE(i < rows_, "matrix access out of range");
-        return row_begin(i);
-    }
-
-    inline Matrix::row_iterator Matrix::operator[](Size i) {
-        return row_begin(i);
-    }
-
-    inline Matrix::row_iterator Matrix::at(Size i) {
-        QL_REQUIRE(i < rows_, "matrix access out of range");
-        return row_begin(i);
-    }
-
-    inline Array Matrix::diagonal() const {
-        Size arraySize = std::min<Size>(rows(), columns());
-        Array tmp(arraySize);
-        for(Size i = 0; i < arraySize; i++)
-            tmp[i] = (*this)[i][i];
-        return tmp;
-    }
-
-    inline Real &Matrix::operator()(Size i, Size j) const {
-        return data_[i*columns()+j];
-    }
```

Comment:
Same, let's at least inline element access.  This means `operator(), `operator[]` and `columns`.